src/H5D.c

```diff
@@ -1143,18 +1143,18 @@ H5Dget_chunk_info_by_coord(hid_t dset_id, const hsize_t *offset, unsigned *filte
  *
  *        typedef int (*H5D_chunk_iter_op_t)(
  *            const hsize_t *offset,
- *            uint32_t filter_mask,
+ *            unsigned filter_mask,
  *            haddr_t addr,
- *            uint32_t size,
+ *            hsize_t size,
  *            void *op_data);
  *
  *      H5D_chunk_iter_op_t parameters:
- *          hsize_t *offset;        IN/OUT: Array of starting logical coordinates of chunk.
- *          uint32_t filter_mask;   IN: Filter mask of chunk.
- *          haddr_t addr;           IN: Offset in file of chunk data.
- *          uint32_t nbytes;        IN: Size in number of bytes of chunk data in file.
- *          void *op_data;          IN/OUT: Pointer to any user-defined data
- *                                  associated with the operation.
+ *          hsize_t *offset;        IN/OUT: Logical position of the chunk’s first element in units of dataset
+ *                                          elements
+ *          unsigned filter_mask;   IN: Bitmask indicating the filters used when the chunk was written haddr_t
```

Comment:
I don't know if clang-format caused this, but looks like the `haddr_t` should be on the following line? as well as `size` missing from the `hsize_t` below.

Reply:
Yes, but we should also fix this in develop.

I'm questioning the need for this documentation here since `H5D_chunk_iter_op_t` is documented in H5Dpublic.h. As far as I can tell the documentation here does not actually appear in doxygen. Why are we documenting this in more than one place?

Reply:
Could we make a decision on whether to fix this as indicated or remove this redundant, unused documentation of `H5D_chunk_iter_op_t` in deference to the existing correct documentation in H5Dpublic.h?

Reply:
I'm going to remove the documentation of `H5D_chunk_iter_op_t` from here also make a PR to remove it from the `develop` branch. Having the redundant documentation is just going to lead to misdocumentation.

Reply:
Fixed in [9f252f1](https://github.com/HDFGroup/hdf5/pull/2329/commits/9f252f193a64ab380db149fb89be7213c8296710) by removing the redundant documentation